### PR TITLE
Make HttpTestServer request handling configurable

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Needed to write a full integration test for https://github.com/crate/crate/issues/14238

### CODE REVIEW

File: server/src/test/java/io/crate/http/HttpTestServer.java [hunk 1]
```diff
@@ -124,8 +127,6 @@ public void shutDown() {
     @ChannelHandler.Sharable
     public class HttpTestServerHandler extends SimpleChannelInboundHandler<Object> {
 
-        private final Logger logger = LogManager.getLogger(HttpTestServerHandler.class.getName());
-
```
Comment (BaurzhanSakhariev) [inline server/src/test/java/io/crate/http/HttpTestServer.java:128]: I removed logger as we can always debug - didn't want to overcomplicate and make it smth like `TriConsumer` only because of logger.

File: server/src/test/java/io/crate/http/HttpTestServer.java [hunk 2]
```diff
@@ -140,29 +141,11 @@ protected void channelRead0(ChannelHandlerContext ctx, Object msg) throws Except
...
 
         private void handleHttpRequest(ChannelHandlerContext ctx, HttpRequest msg) throws UnsupportedEncodingException {
-            String uri = msg.uri();
-            QueryStringDecoder decoder = new QueryStringDecoder(uri);
-            logger.debug("Got Request for " + uri);
             HttpResponseStatus status = fail ? HttpResponseStatus.BAD_REQUEST : HttpResponseStatus.OK;
-
             ByteArrayOutputStream out = new ByteArrayOutputStream();
             try {
                 JsonGenerator generator = JSON_FACTORY.createGenerator(out, JsonEncoding.UTF8);
```
Comment (seut) [inline server/src/test/java/io/crate/http/HttpTestServer.java:147]: Why not creating the generator at the caller side? It's not used here afaik. I'd suggest to pass the output stream instead, this would allow handlers to respond whatever content they want.

File: server/src/test/java/io/crate/http/HttpTestServer.java [hunk 3]
```diff
@@ -176,7 +159,6 @@ private void handleHttpRequest(ChannelHandlerContext ctx, HttpRequest msg) throw
 
         @Override
         public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
-            logger.warn("Unexpected exception from downstream.", cause);
```
Comment (seut) [inline server/src/test/java/io/crate/http/HttpTestServer.java:179]: How is this change related?

**Review by seut — APPROVED** [review]
Usually, I'd prefer doing such changes with the same commit which are requiring it. This would make the need to the changed more clear.

File: server/src/test/java/io/crate/http/HttpTestServer.java (hunk 3 above)
Comment (BaurzhanSakhariev) [reply server/src/test/java/io/crate/http/HttpTestServer.java:179]: got the logger back

**Comment by BaurzhanSakhariev** [comment]
Unrelated timeouts
https://jenkins.crate.io/job/CrateDB/job/crate_on_pr/12989/testReport/junit/io.crate.integrationtests/SQLTypeMappingTest/test_dynamic_null_array_get_by_pk/

retest this please

File: server/src/test/java/io/crate/http/HttpTestServer.java (hunk 2 above)
Comment (seut) [reply server/src/test/java/io/crate/http/HttpTestServer.java:154]: > Then we would need to repeat JSON_FACTORY configuration code in all callers

yep, but given that it's one line, I don't see a big issue.
But also fine to keep it like that, no strong opinion.

**Review by seut — APPROVED** [review]
